src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt

```diff
@@ -349,6 +354,31 @@ abstract class FixupSession(
           }
         }
 
+        // Skip the formatting if these edits are part of an undo operation.
+        if (!undoInProgress) {
+          // It is safe to use selectionRange here, because it is set in the constructor
+          // by calling triggerFixupAsync() which in turn calls ensureSelectionRange().
+          // The value of this property does not change
+          // when the user selects a different snippet in the UI.
+          // FIXME this causes a problem when the entire file becomes shorter than the original
+          // selection range.
+          selectionRange!!.let {
```

Comment:
Please do not use !!

Reply:
I believe the `!!` is justified here. However, I just realised that this FIXME still has to be fixed...